packages/esm-ugandaemr-app/src/pages/system-info/system-info.component.tsx

```diff
@@ -142,47 +135,52 @@ const SystemInfoPage = () => {
   const [buildInfo, setBuildInfo] = useState({});
   const [emrVersion, setEMRVersion] = useState('4.0');
   const { systemInfo, isError, isLoading } = useGetSystemInformation();
-  const [facilityCodeDetails, setFacilityCodeDetails] = useState<FacilityCodeDetails>({ value: null, uuid: null });
+  const [facilityCodeDetails, setFacilityCodeDetails] = useState<FacilityCodeDetails>({ value: null });
 
   const { facilityIds } = useRetrieveFacilityCode();
 
   useEffect(() => {
     if (facilityIds && facilityIds.length) {
       setFacilityCodeDetails({
-        ...facilityCodeDetails,
         value: facilityIds[0]['value'],
-        uuid: facilityIds[0]['uuid'],
       });
     }
-  }, [facilityCodeDetails, facilityIds]);
+  }, [facilityIds]);
 
-  useEffect(() => {
-    if (facilityCodeDetails.uuid !== null && facilityCodeDetails.value !== null) {
-      updatePropertyValue(facilityCodeDetails.uuid, facilityCodeDetails.value).then(
+  const updateFacilityCode = useCallback(() => {
+    if (facilityCodeDetails.value) {
+      updatePropertyValue('ugandaemrsync.national.health.facility.registry.identifier', facilityCodeDetails.value).then(
```

Comment:
this too

Reply:
> this too

This has also been addressed in the subsequent commit